### PR TITLE
cloud-data-discoverer Improve error handling 

### PR DESCRIPTION
**Pull request by borg-z**
## Description

During cluster bootstrap process, we initiate the cloud-data-discoverer service, which identifies cloud resources and their parameters. Based on this data, we configure the cluster and create resources such as StorageClass. The cloud-data-discoverer executes several steps, making multiple attempts for each step. If all attempts fail, an error is logged. 
And this is where the problem begins. It simply logs the error, exits the loop, and reports "Finish data discovery."

- Increased the number of retry attempts.
- small logging fix
~~- Added a Fatalln to ensure the process fully restarts if all attempts fail.~~

> As mentioned in the comment below this is not the best solution, so just increased the number of attempts.

logs before this fix:

```
{"level":"error","msg":"Attempt 1 of 3. Cannot update cloud data resource: resource name may not be empty","time":"2024-09-02T13:30:40Z"}
{"level":"info","msg":"Waiting 3 seconds before next attempt","time":"2024-09-02T13:30:40Z"}
{"level":"error","msg":"Attempt 2 of 3. Cannot update cloud data resource: resource name may not be empty","time":"2024-09-02T13:30:43Z"}
{"level":"info","msg":"Waiting 3 seconds before next attempt","time":"2024-09-02T13:30:43Z"}
{"level":"error","msg":"Attempt 3 of 3. Cannot update cloud data resource: resource name may not be empty","time":"2024-09-02T13:30:46Z"}
{"level":"error","msg":"Cannot update cloud data resource. Timed out. See error messages below.","time":"2024-09-02T13:30:46Z"}
{"level":"info","msg":"Finish cloud data discovery step","time":"2024-09-02T13:30:46Z"}
{"level":"info","msg":"Start orphaned disks discovery step","time":"2024-09-02T13:30:46Z"}
{"level":"info","msg":"No disks found","time":"2024-09-02T13:30:49Z"}
{"level":"info","msg":"Finish orphaned disks discovery step","time":"2024-09-02T13:30:49Z"}
{"level":"info","msg":"Finish data discovery","time":"2024-09-02T13:30:49Z"}
```

## Why do we need it, and what problem does it solve?

Currently, during the  bootstrap process, there can be situations where the cloud-data-discoverer fails to retrieve data and does not make attempts to do so. As a result, critical resources such as StorageClass and others are not created.

## What is the expected result?

The cloud-data-discoverer continues its operation despite temporary issues within the cluster.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: go_lib
type: fix
summary: cloud-data-discoverer continues its operation despite temporary issues within the cluster.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by borg-z** [comment]
```
kubectl -n d8-monitoring get pvc
NAME                                           STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
prometheus-longterm-db-prometheus-longterm-0   Bound    pvc-86da4853-8a51-4aa2-88bd-9e6c75cc6fcf   40Gi       RWO            ceph-hdd       30s
prometheus-main-db-prometheus-main-0           Bound    pvc-880031df-3e2f-4fef-b3bf-4e519922fc3b   40Gi       RWO            ceph-hdd       29s
```

File: go_lib/cloud-data/reconciler.go [hunk 1]
```diff
@@ -262,7 +262,7 @@ func (c *Reconciler) instanceTypesReconcile(ctx context.Context) {
 
 	if err != nil {
 		c.updateResourceErrorMetric.WithLabelValues().Set(1.0)
-		c.logger.Errorln("Cannot update cloud data resource. Timed out. See error messages below.")
+		c.logger.Fatalln("Cannot update cloud data resource. Timed out. See error messages below.")
```
Comment (name212) [inline go_lib/cloud-data/reconciler.go:265]: On the one hand, the decision to crash is a good one, a kube one. But it was done on purpose. Some providers, especially with openstack, like to block the API at a certain threshold of incorrect requests, so we always do not try to discover data from the cloud, do fire the alert and leave the decision to restart to the engineer.

In fact, a good solution would be to crash on requests to the kube-api, and fire the alert on requests to the cloud

**Review by name212 — CHANGES_REQUESTED** [review]
suggestion

**Review by aleksey-su — APPROVED** [review]

**Review by name212 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/openstack'...

**Comment by name212** [comment]
/e2e/abort openstack;Standard;containerd;1.27 10902016956 10902016956-1-con-1-27 /sys/deckhouse-oss/install:pr9570

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/openstack'...

**Review by name212 — APPROVED** [review]

**Pull request merged**